<a href="https://colab.research.google.com/github/graciouschinaza/weather-Map/blob/main/weather_Map.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import requests
import pandas as pd

In [5]:
import requests
import pandas as pd

api_key = "8772f358479def3944d766d2a6c350e4"

cities = ["Aba", "Onitsha", "Warri"]

weather_data = []

for city in cities:
    url = f"https://api.openweathermap.org/data/2.5/weather?q={city}&appid={api_key}&units=metric"

    response = requests.get(url)

    if response.status_code == 200:
        data = response.json()

        weather_data.append({
            "City": data["name"],
            "Temperature": data["main"]["temp"],
            "Humidity": data["main"]["humidity"],
            "Weather Condition": data["weather"][0]["description"],
            "Wind Speed": data["wind"]["speed"],
            "Date and Time": pd.to_datetime(data["dt"], unit="s")
        })
    else:
        print(f"Could not retrieve data for {city}")

weather_data

[{'City': 'Aba',
  'Temperature': 25.89,
  'Humidity': 83,
  'Weather Condition': 'moderate rain',
  'Wind Speed': 3.26,
  'Date and Time': Timestamp('2026-08-10 14:44:41')},
 {'City': 'Onitsha',
  'Temperature': 26.06,
  'Humidity': 86,
  'Weather Condition': 'light rain',
  'Wind Speed': 2.59,
  'Date and Time': Timestamp('2026-08-10 14:45:58')},
 {'City': 'Warri',
  'Temperature': 26.87,
  'Humidity': 75,
  'Weather Condition': 'light rain',
  'Wind Speed': 4.66,
  'Date and Time': Timestamp('2026-08-10 14:45:28')}]

In [6]:
df = pd.DataFrame(weather_data)

df

,City,Temperature,Humidity,Weather Condition,Wind Speed,Date and Time
0,Aba,25.89,83,moderate rain,3.26,2026-08-10 14:44:41
1,Onitsha,26.06,86,light rain,2.59,2026-08-10 14:45:58
2,Warri,26.87,75,light rain,4.66,2026-08-10 14:45:28


In [9]:
df.shape

(3, 6)

In [10]:
df.columns

Index(['City', 'Temperature', 'Humidity', 'Weather Condition', 'Wind Speed',
       'Date and Time'],
      dtype='object')

In [13]:
df.dtypes

,0
City,object
Temperature,float64
Humidity,int64
Weather Condition,object
Wind Speed,float64
Date and Time,datetime64[ns]


In [14]:
df.isnull().sum()

,0
City,0
Temperature,0
Humidity,0
Weather Condition,0
Wind Speed,0
Date and Time,0


no missing values

In [15]:
df.duplicated().sum()

np.int64(0)

no duplicate rows

In [18]:
df["Date and Time"] = pd.to_datetime(df["Date and Time"])

In [19]:
df = df.rename(columns={
    "City": "City",
    "Temperature": "Temperature_C",
    "Humidity": "Humidity_Percent",
    "Weather Condition": "Weather_Condition",
    "Wind Speed": "Wind_Speed",
    "Date and Time": "Date_Time"
})

In [20]:
df.head()

,City,Temperature_C,Humidity_Percent,Weather_Condition,Wind_Speed,Date_Time
0,Aba,25.89,83,moderate rain,3.26,2026-08-10 14:44:41
1,Onitsha,26.06,86,light rain,2.59,2026-08-10 14:45:58
2,Warri,26.87,75,light rain,4.66,2026-08-10 14:45:28


In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3 entries, 0 to 2
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   City               3 non-null      object        
 1   Temperature_C      3 non-null      float64       
 2   Humidity_Percent   3 non-null      int64         
 3   Weather_Condition  3 non-null      object        
 4   Wind_Speed         3 non-null      float64       
 5   Date_Time          3 non-null      datetime64[ns]
dtypes: datetime64[ns](1), float64(2), int64(1), object(2)
memory usage: 276.0+ bytes


In [22]:
df

,City,Temperature_C,Humidity_Percent,Weather_Condition,Wind_Speed,Date_Time
0,Aba,25.89,83,moderate rain,3.26,2026-08-10 14:44:41
1,Onitsha,26.06,86,light rain,2.59,2026-08-10 14:45:58
2,Warri,26.87,75,light rain,4.66,2026-08-10 14:45:28


In [23]:
df.to_csv("weather_data_cleaned.csv",index=False)

In [24]:
import os
os.path.exists("weather_data_cleaned.csv")

True

In [25]:
check_data=pd.read_csv("weather_data_cleaned.csv")
check_data

,City,Temperature_C,Humidity_Percent,Weather_Condition,Wind_Speed,Date_Time
0,Aba,25.89,83,moderate rain,3.26,2026-08-10 14:44:41
1,Onitsha,26.06,86,light rain,2.59,2026-08-10 14:45:58
2,Warri,26.87,75,light rain,4.66,2026-08-10 14:45:28


In [26]:
from google.colab import files
files.download("weather_data_cleaned.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [27]:
df

,City,Temperature_C,Humidity_Percent,Weather_Condition,Wind_Speed,Date_Time
0,Aba,25.89,83,moderate rain,3.26,2026-08-10 14:44:41
1,Onitsha,26.06,86,light rain,2.59,2026-08-10 14:45:58
2,Warri,26.87,75,light rain,4.66,2026-08-10 14:45:28


In [29]:
temperature_comparison=df[["City","Temperature_C"]]
temperature_comparison

,City,Temperature_C
0,Aba,25.89
1,Onitsha,26.06
2,Warri,26.87


Temperature Comparison

In [30]:
hottest_city = df.loc[df["Temperature_C"].idxmax()]

print("Hottest City:", hottest_city["City"])
print("Temperature:", hottest_city["Temperature_C"], "°C")

Hottest City: Warri
Temperature: 26.87 °C


Warri is the hottest city with the Temperature as 26.87 °C

In [31]:
coolest_city = df.loc[df["Temperature_C"].idxmin()]

print("Coolest City:", coolest_city["City"])
print("Temperature:", coolest_city["Temperature_C"], "°C")

Coolest City: Aba
Temperature: 25.89 °C


Aba is the coolest city with its Temperature as 25.89 °C

In [32]:
highest_humidity = df.loc[df["Humidity_Percent"].idxmax()]

print("City with Highest Humidity:", highest_humidity["City"])
print("Humidity:", highest_humidity["Humidity_Percent"], "%")

City with Highest Humidity: Onitsha
Humidity: 86 %


Onitsha is the city with the hightest humidity among the 3 cities

In [33]:
weather_comparison = df[["City", "Weather_Condition"]]

weather_comparison

,City,Weather_Condition
0,Aba,moderate rain
1,Onitsha,light rain
2,Warri,light rain


In [34]:
hottest = df.loc[df["Temperature_C"].idxmax()]
coolest = df.loc[df["Temperature_C"].idxmin()]
most_humid = df.loc[df["Humidity_Percent"].idxmax()]

print("WEATHER ANALYSIS SUMMARY")
print("------------------------")
print(f"Hottest city: {hottest['City']} ({hottest['Temperature_C']} °C)")
print(f"Coolest city: {coolest['City']} ({coolest['Temperature_C']} °C)")
print(f"Highest humidity: {most_humid['City']} ({most_humid['Humidity_Percent']}%)")
print("\nWeather conditions:")
print(df[["City", "Weather_Condition"]].to_string(index=False))

WEATHER ANALYSIS SUMMARY
------------------------
Hottest city: Warri (26.87 °C)
Coolest city: Aba (25.89 °C)
Highest humidity: Onitsha (86%)

Weather conditions:
   City Weather_Condition
    Aba     moderate rain
Onitsha        light rain
  Warri        light rain
